### PR TITLE
Log extra information before dropping connection

### PR DESCRIPTION
**Pull request by nirbhayc**
For debugging purpose.

### CODE REVIEW

File: plugins/newsql/newsql.c [hunk 1]
```diff
@@ -2191,8 +2203,10 @@ static int handle_newsql_request(comdb2_appsock_arg_t *arg)
 
     CDB2SQLQUERY *sql_query = query->sqlquery;
 
-    if (!clnt.admin && do_query_on_master_check(dbenv, &clnt, sql_query))
+    if (!clnt.admin && do_query_on_master_check(dbenv, &clnt, sql_query)) {
+        logmsg(LOGMSG_ERROR, "query on master, will be rejected\n");
```
Comment (riverszhang89) [inline plugins/newsql/newsql.c]: This will cause log spewing.

File: plugins/newsql/newsql.c [hunk 2]
```diff
@@ -1876,15 +1879,18 @@ static CDB2QUERY *read_newsql_query(struct dbenv *dbenv,
                          gbl_nid_dbname, NULL, 0, 0) != 1) {
             newsql_error(clnt, "Client certificate authentication failed.",
                          CDB2ERR_CONNECT_ERROR);
+            logmsg(LOGMSG_ERROR, "client certificate authentication failed\n");
```
Comment (riverszhang89) [inline plugins/newsql/newsql.c]: No need to log this down. `sslio_accept` already does it.

File: plugins/newsql/newsql.c [hunk 3]
```diff
@@ -1989,9 +1995,14 @@ static CDB2QUERY *read_newsql_query(struct dbenv *dbenv,
...
 
-    if (!query) return NULL;
+    if (!query) {
+        logmsg(LOGMSG_ERROR, "query missing\n");
+        return NULL;
+    }
+
     if (errno != 0) {
         cdb2__query__free_unpacked(query, &pb_alloc);
+        logmsg(LOGMSG_ERROR, "errno: %d\n", errno);
```
Comment (riverszhang89) [inline plugins/newsql/newsql.c]: A bit more context would make it clearer.

File: plugins/newsql/newsql.c [hunk 4]
```diff
@@ -2359,8 +2375,10 @@ static int handle_newsql_request(comdb2_appsock_arg_t *arg)
                 srs_tran_destroy(&clnt);
         }
 
-        if (rc && !clnt.in_client_trans)
+        if (rc && !clnt.in_client_trans) {
+            logmsg(LOGMSG_ERROR, "rc = %d\n", rc);
```
Comment (riverszhang89) [inline plugins/newsql/newsql.c]: Need a bit more context.

File: plugins/newsql/newsql.c [hunk 5]
```diff
@@ -2166,6 +2177,7 @@ static int handle_newsql_request(comdb2_appsock_arg_t *arg)
         }
         newsql_error(&clnt, "Exhausted appsock connections.",
                      CDB2__ERROR_CODE__APPSOCK_LIMIT);
+        logmsg(LOGMSG_ERROR, "exhaused appsock connections\n");
```
Comment (mistachkin) [inline plugins/newsql/newsql.c]: Typo here, should be 'exhausted'.

File: plugins/newsql/newsql.c (hunk 1 above)
Comment (nirbhayc) [reply plugins/newsql/newsql.c]: This PR is to help me track an issue with a specific db. I am not planning to merge it into the trunk.

File: plugins/newsql/newsql.c (hunk 2 above)
Comment (nirbhayc) [reply plugins/newsql/newsql.c]: sslio_accept passes NULL for error (out) string. Also, don't see sslio_accept_or_connect() logging any info.

File: plugins/newsql/newsql.c (hunk 5 above)
Comment (nirbhayc) [reply plugins/newsql/newsql.c]: Fixed

File: plugins/newsql/newsql.c (hunk 4 above)
Comment (nirbhayc) [reply plugins/newsql/newsql.c]: Done

File: plugins/newsql/newsql.c (hunk 3 above)
Comment (nirbhayc) [reply plugins/newsql/newsql.c]: Done

File: plugins/newsql/newsql.c (hunk 1 above)
Comment (nirbhayc) [reply plugins/newsql/newsql.c]: But, if you think these are useful enough, I can add them with lower log level (debug).

**Comment by nirbhayc** [comment]
I used this PR to track an issue.

**Pull request closed without merging**